// @unzer/nodejs-sdk/dist/lib/payments/types/AbstractPaymentType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AbstractPaymentType = /** @class */ (function () {
    function AbstractPaymentType() {
    }
    /**
     * Set Unzer instance
     *
     * @param {Unzer} unzer
     */
    AbstractPaymentType.prototype.setUnzer = function (unzer) {
        this._unzer = unzer;
    };
    /**
     * Get instance Unzer
     *
     * @returns {Unzer}
     */
    AbstractPaymentType.prototype.getUnzer = function () {
        return this._unzer;
    };
    /**
     * Set Payment Id
     *
     * @param {string} paymentId
     */
    AbstractPaymentType.prototype.setId = function (paymentId) {
        this._id = paymentId;
    };
    /**
     * Get Payment Id
     *
     * @returns {string}
     */
    AbstractPaymentType.prototype.getId = function () {
        return this._id;
    };
    /**
     * Authorize with payment card
     *
     * @param {authorizeObject} args
     * @returns {Promise<Authorization>}
     */
    AbstractPaymentType.prototype.authorize = function (args) {
        return this.getUnzer().authorize(args);
    };
    /**
     * Charge on authorize
     *
     * @param {chargeAuthorizeObject} args
     * @returns {Promise<Charge>}
     */
    AbstractPaymentType.prototype.charge = function (args) {
        return this.getUnzer().charge(args);
    };
    /**
     * Get geoLocation
     *
     * @returns {any}
     */
    AbstractPaymentType.prototype.getGeoLocation = function () {
        return this._geoLocation;
    };
    /**
     * Set geoLocation
     *
     * @param {any} geoLocation
     * @returns {any}
     */
    AbstractPaymentType.prototype.setGeoLocation = function (geoLocation) {
        this._geoLocation = geoLocation;
    };
    return AbstractPaymentType;
}());
exports.default = AbstractPaymentType;
