// @unzer/nodejs-sdk/dist/lib/payments/types/InstallmentSecuredPlan.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var apiURL = require("../../configs/ApiUrls");
var AbstractPaymentType_1 = require("./AbstractPaymentType");
var InstallmentSecuredPlan = /** @class */ (function (_super) {
    __extends(InstallmentSecuredPlan, _super);
    function InstallmentSecuredPlan() {
        return _super.call(this) || this;
    }
    /**
     * Set Number of Rate
     *
     * @param {string} numberOfRates
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setNumberOfRates = function (numberOfRates) {
        this._numberOfRates = numberOfRates;
        return this;
    };
    /**
     * Get Number of Rate
     *
     * @returns {string}
     */
    InstallmentSecuredPlan.prototype.getNumberOfRates = function () {
        return this._numberOfRates;
    };
    /**
     * Set Day of purchase
     *
     * @param {string} dayOfPurchase
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setDayOfPurchase = function (dayOfPurchase) {
        this._dayOfPurchase = dayOfPurchase;
        return this;
    };
    /**
     * Get Day of Purchase
     *
     * @returns {string}
     */
    InstallmentSecuredPlan.prototype.getDayOfPurchase = function () {
        return this._dayOfPurchase;
    };
    /**
     * Set Order Date
     *
     * @param {string} orderDate
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setOrderDate = function (orderDate) {
        this._orderDate = orderDate;
        return this;
    };
    /**
     * Get Order Date
     *
     * @returns {string}
     */
    InstallmentSecuredPlan.prototype.getOrderDate = function () {
        return this._orderDate;
    };
    /**
     * Set Total Purchase Amount
     *
     * @param {number} totalPurchaseAmount
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setTotalPurchaseAmount = function (totalPurchaseAmount) {
        this._totalPurchaseAmount = totalPurchaseAmount;
        return this;
    };
    /**
     * Get Total Purchase Amount
     *
     * @returns {number}
     */
    InstallmentSecuredPlan.prototype.getTotalPurchaseAmount = function () {
        return this._totalPurchaseAmount;
    };
    /**
     * Set Interest Amount
     *
     * @param {number} totalInterestAmount
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setTotalInterestAmount = function (totalInterestAmount) {
        this._totalInterestAmount = totalInterestAmount;
        return this;
    };
    /**
     * Get Interest Amount
     *
     * @returns {number}
     */
    InstallmentSecuredPlan.prototype.getTotalInterestAmount = function () {
        return this._totalInterestAmount;
    };
    /**
     * Set Total Amount
     *
     * @param {number} totalAmount
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setTotalAmount = function (totalAmount) {
        this._totalAmount = totalAmount;
        return this;
    };
    /**
     * Get Total Amount
     *
     * @returns {number}
     */
    InstallmentSecuredPlan.prototype.getTotalAmount = function () {
        return this._totalAmount;
    };
    /**
     * Set Effective Interest Rate
     *
     * @param {number} effectiveInterestRate
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setEffectiveInterestRate = function (effectiveInterestRate) {
        this._effectiveInterestRate = effectiveInterestRate;
        return this;
    };
    /**
     * Get Total Amount
     *
     * @returns {number}
     */
    InstallmentSecuredPlan.prototype.getEffectiveInterestRate = function () {
        return this._effectiveInterestRate;
    };
    /**
     * Set Nominal Interest Rate
     *
     * @param {number} nominalInterestRate
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setNominalInterestRate = function (nominalInterestRate) {
        this._nominalInterestRate = nominalInterestRate;
        return this;
    };
    /**
     * Get Nominal Interest Rate
     *
     * @returns {number}
     */
    InstallmentSecuredPlan.prototype.getNominalInterestRate = function () {
        return this._nominalInterestRate;
    };
    /**
     * Set Fee First Rate
     *
     * @param {number} feeFirstRate
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setFeeFirstRate = function (feeFirstRate) {
        this._feeFirstRate = feeFirstRate;
        return this;
    };
    /**
     * Get Nominal Interest Rate
     *
     * @returns {number}
     */
    InstallmentSecuredPlan.prototype.getFeeFirstRate = function () {
        return this._feeFirstRate;
    };
    /**
     * Set Fee Per Rate
     *
     * @param {number} feePerRate
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setFeePerRate = function (feePerRate) {
        this._feePerRate = feePerRate;
        return this;
    };
    /**
     * Get Nominal Interest Rate
     *
     * @returns {number}
     */
    InstallmentSecuredPlan.prototype.getFeePerRate = function () {
        return this._feePerRate;
    };
    /**
     * Set Monthly Rate
     *
     * @param {number} monthlyRate
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setMonthlyRate = function (monthlyRate) {
        this._monthlyRate = monthlyRate;
        return this;
    };
    /**
     * Get Monthly Rate
     *
     * @returns {number}
     */
    InstallmentSecuredPlan.prototype.getMonthlyRate = function () {
        return this._monthlyRate;
    };
    /**
     * Set Last Rate
     *
     * @param {number} lastRate
     * @returns {InstallmentSecuredPlan}
     */
    InstallmentSecuredPlan.prototype.setLastRate = function (lastRate) {
        this._lastRate = lastRate;
        return this;
    };
    /**
     * Get Monthly Rate
     *
     * @returns {number}
     */
    InstallmentSecuredPlan.prototype.getLastRate = function () {
        return this._lastRate;
    };
    /**
     * Get url end point
     *
     * @returns {string}
     */
    InstallmentSecuredPlan.prototype.getTypeUrl = function () {
        return apiURL.URL_TYPE_INSTALLMENT_SECURED_PLANS;
    };
    /**
     * Get Payload
     *
     * @returns {*}
     */
    InstallmentSecuredPlan.prototype.getPayload = function () {
        return {};
    };
    return InstallmentSecuredPlan;
}(AbstractPaymentType_1.default));
exports.default = InstallmentSecuredPlan;
