// @unzer/nodejs-sdk/dist/lib/payments/types/InstallmentSecured.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var apiURL = require("../../configs/ApiUrls");
var AbstractPaymentType_1 = require("./AbstractPaymentType");
var InstallmentSecured = /** @class */ (function (_super) {
    __extends(InstallmentSecured, _super);
    function InstallmentSecured() {
        return _super.call(this) || this;
    }
    /**
     * Set Iban
     *
     * @param {string} iban
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setIban = function (iban) {
        this._iban = iban;
        return this;
    };
    /**
     * Get Iban
     *
     * @returns {string}
     */
    InstallmentSecured.prototype.getIban = function () {
        return this._iban;
    };
    /**
     * Set BIC
     *
     * @param {string} bic
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setBic = function (bic) {
        this._bic = bic;
        return this;
    };
    /**
     * Get Bic
     *
     * @returns {string}
     */
    InstallmentSecured.prototype.getBic = function () {
        return this._bic;
    };
    /**
     * Set Account Holader
     *
     * @param {string} accountHolder
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setAccountHolder = function (accountHolder) {
        this._accountHolder = accountHolder;
        return this;
    };
    /**
     * Get Account Holder
     *
     * @returns {string}
     */
    InstallmentSecured.prototype.getAccountHolder = function () {
        return this._accountHolder;
    };
    /**
     * Set Invoice Date
     *
     * @param {string} invoiceDate
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setInvoiceDate = function (invoiceDate) {
        this._invoiceDate = invoiceDate;
        return this;
    };
    /**
     * Get Invoice Date
     *
     * @returns {string}
     */
    InstallmentSecured.prototype.getInvoiceDate = function () {
        return this._invoiceDate;
    };
    /**
     * Set Invoice Due Date
     *
     * @param {string} invoiceDueDate
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setInvoiceDueDate = function (invoiceDueDate) {
        this._invoiceDueDate = invoiceDueDate;
        return this;
    };
    /**
     * Get Invoice Due Date
     *
     * @returns {string}
     */
    InstallmentSecured.prototype.getInvoiceDueDate = function () {
        return this._invoiceDueDate;
    };
    /**
     * Set Number of Rate
     *
     * @param {string} numberOfRates
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setNumberOfRates = function (numberOfRates) {
        this._numberOfRates = numberOfRates;
        return this;
    };
    /**
     * Get Number of Rate
     *
     * @returns {string}
     */
    InstallmentSecured.prototype.getNumberOfRates = function () {
        return this._numberOfRates;
    };
    /**
     * Set Day of purchase
     *
     * @param {string} dayOfPurchase
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setDayOfPurchase = function (dayOfPurchase) {
        this._dayOfPurchase = dayOfPurchase;
        return this;
    };
    /**
     * Get Day of Purchase
     *
     * @returns {string}
     */
    InstallmentSecured.prototype.getDayOfPurchase = function () {
        return this._dayOfPurchase;
    };
    /**
     * Set Order Date
     *
     * @param {string} orderDate
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setOrderDate = function (orderDate) {
        this._orderDate = orderDate;
        return this;
    };
    /**
     * Get Order Date
     *
     * @returns {string}
     */
    InstallmentSecured.prototype.getOrderDate = function () {
        return this._orderDate;
    };
    /**
     * Set Total Purchase Amount
     *
     * @param {number} totalPurchaseAmount
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setTotalPurchaseAmount = function (totalPurchaseAmount) {
        this._totalPurchaseAmount = totalPurchaseAmount;
        return this;
    };
    /**
     * Get Total Purchase Amount
     *
     * @returns {number}
     */
    InstallmentSecured.prototype.getTotalPurchaseAmount = function () {
        return this._totalPurchaseAmount;
    };
    /**
     * Set Interest Amount
     *
     * @param {number} totalInterestAmount
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setTotalInterestAmount = function (totalInterestAmount) {
        this._totalInterestAmount = totalInterestAmount;
        return this;
    };
    /**
     * Get Interest Amount
     *
     * @returns {number}
     */
    InstallmentSecured.prototype.getTotalInterestAmount = function () {
        return this._totalInterestAmount;
    };
    /**
     * Set Total Amount
     *
     * @param {number} totalAmount
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setTotalAmount = function (totalAmount) {
        this._totalAmount = totalAmount;
        return this;
    };
    /**
     * Get Total Amount
     *
     * @returns {number}
     */
    InstallmentSecured.prototype.getTotalAmount = function () {
        return this._totalAmount;
    };
    /**
     * Set Effective Interest Rate
     *
     * @param {number} effectiveInterestRate
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setEffectiveInterestRate = function (effectiveInterestRate) {
        this._effectiveInterestRate = effectiveInterestRate;
        return this;
    };
    /**
     * Get Effective Interest Rate
     *
     * @returns {number}
     */
    InstallmentSecured.prototype.getEffectiveInterestRate = function () {
        return this._effectiveInterestRate;
    };
    /**
     * Set Nominal Interest Rate
     *
     * @param {number} nominalInterestRate
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setNominalInterestRate = function (nominalInterestRate) {
        this._nominalInterestRate = nominalInterestRate;
        return this;
    };
    /**
     * Get Nominal Interest Rate
     *
     * @returns {number}
     */
    InstallmentSecured.prototype.getNominalInterestRate = function () {
        return this._nominalInterestRate;
    };
    /**
     * Set Fee First Rate
     *
     * @param {number} feeFirstRate
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setFeeFirstRate = function (feeFirstRate) {
        this._feeFirstRate = feeFirstRate;
        return this;
    };
    /**
     * Get Nominal Interest Rate
     *
     * @returns {number}
     */
    InstallmentSecured.prototype.getFeeFirstRate = function () {
        return this._feeFirstRate;
    };
    /**
     * Set Fee Per Rate
     *
     * @param {number} feePerRate
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setFeePerRate = function (feePerRate) {
        this._feePerRate = feePerRate;
        return this;
    };
    /**
     * Get Nominal Interest Rate
     *
     * @returns {number}
     */
    InstallmentSecured.prototype.getFeePerRate = function () {
        return this._feePerRate;
    };
    /**
     * Set Monthly Rate
     *
     * @param {number} monthlyRate
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setMonthlyRate = function (monthlyRate) {
        this._monthlyRate = monthlyRate;
        return this;
    };
    /**
     * Get Monthly Rate
     *
     * @returns {number}
     */
    InstallmentSecured.prototype.getMonthlyRate = function () {
        return this._monthlyRate;
    };
    /**
     * Set Last Rate
     *
     * @param {number} lastRate
     * @returns {InstallmentSecured}
     */
    InstallmentSecured.prototype.setLastRate = function (lastRate) {
        this._lastRate = lastRate;
        return this;
    };
    /**
     * Get Monthly Rate
     *
     * @returns {number}
     */
    InstallmentSecured.prototype.getLastRate = function () {
        return this._lastRate;
    };
    /**
     * Get url end point
     *
     * @returns {string}
     */
    InstallmentSecured.prototype.getTypeUrl = function () {
        return apiURL.URL_TYPE_INSTALLMENT_SECURED;
    };
    /**
     * Get Payload
     *
     * @returns {*}
     */
    InstallmentSecured.prototype.getPayload = function () {
        return {
            iban: this.getIban(),
            bic: this.getBic(),
            accountHolder: this.getAccountHolder(),
            invoiceDate: this.getInvoiceDate(),
            invoiceDueDate: this.getInvoiceDueDate(),
            numberOfRates: this.getNumberOfRates(),
            dayOfPurchase: this.getDayOfPurchase(),
            totalPurchaseAmount: this.getTotalPurchaseAmount(),
            totalInterestAmount: this.getTotalInterestAmount(),
            totalAmount: this.getTotalAmount(),
            effectiveInterestRate: this.getEffectiveInterestRate(),
            nominalInterestRate: this.getNominalInterestRate(),
            feeFirstRate: this.getFeeFirstRate(),
            feePerRate: this.getFeePerRate(),
            monthlyRate: this.getMonthlyRate(),
            lastRate: this.getLastRate(),
        };
    };
    return InstallmentSecured;
}(AbstractPaymentType_1.default));
exports.default = InstallmentSecured;
