// @unzer/nodejs-sdk/dist/lib/payments/types/InvoiceSecured.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var apiURL = require("../../configs/ApiUrls");
var AbstractPaymentType_1 = require("./AbstractPaymentType");
var InvoiceSecured = /** @class */ (function (_super) {
    __extends(InvoiceSecured, _super);
    function InvoiceSecured() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Get url end point
     *
     * @returns {string}
     */
    InvoiceSecured.prototype.getTypeUrl = function () {
        return apiURL.URL_TYPE_INVOICE_SECURED;
    };
    /**
     * Set Invoice Id
     *
     * @param {string} invoiceId
     */
    InvoiceSecured.prototype.setInvoiceId = function (invoiceId) {
        this._invoiceId = invoiceId;
    };
    /**
     * Get Invoice Id
     *
     * @returns {string}
     */
    InvoiceSecured.prototype.getInvoiceId = function () {
        return this._invoiceId;
    };
    /**
     * Get Payload
     *
     * @returns
     */
    InvoiceSecured.prototype.getPayload = function () {
        if (typeof this.getInvoiceId() === 'undefined') {
            return {};
        }
        return {
            invoiceId: this._invoiceId
        };
    };
    return InvoiceSecured;
}(AbstractPaymentType_1.default));
exports.default = InvoiceSecured;
