// @unzer/nodejs-sdk/dist/lib/payments/business/AbstractPayment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AbstractPayment = /** @class */ (function () {
    function AbstractPayment(unzer) {
        this._unzer = unzer;
        this._id = '';
    }
    /**
     * Set a payment object
     *
     * @param {Payment} payment
     */
    AbstractPayment.prototype.setPayment = function (payment) {
        this._payment = payment;
    };
    /**
     * Get payment object
     *
     * @returns {Payment}
     */
    AbstractPayment.prototype.getPayment = function () {
        return this._payment;
    };
    /**
     * Get instance Unzer
     *
     * @returns {Unzer}
     */
    AbstractPayment.prototype.getUnzer = function () {
        return this._unzer;
    };
    /**
     * Set Payment Id
     *
     * @param {string} paymentId
     */
    AbstractPayment.prototype.setId = function (paymentId) {
        this._id = paymentId;
    };
    /**
     * Get Payment Id
     *
     * @returns {string}
     */
    AbstractPayment.prototype.getId = function () {
        return this._id;
    };
    /**
     * Get payload object
     *
     * @returns {*}
     */
    AbstractPayment.prototype.getPayload = function () {
        return this._payload;
    };
    /**
     * Set payload object
     *
     * @param {*} payload
     * @returns
     */
    AbstractPayment.prototype.setPayload = function (payload) {
        return this._payload = payload;
    };
    return AbstractPayment;
}());
exports.default = AbstractPayment;
