// @unzer/nodejs-sdk/dist/lib/configs/ApiUrls.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.URL_LINKPAY_AUTHORIZE = exports.URL_LINKPAY_CHARGE = exports.URL_LINKPAY = exports.URL_WEBHOOK = exports.URL_PAYPAGE_AUTHORIZE = exports.URL_PAYPAGE_CHARGE = exports.URL_TYPE_INSTALLMENT_SECURED_PLANS = exports.URL_TYPE_INSTALLMENT_SECURED = exports.URL_TYPE_WECHATPAY = exports.URL_TYPE_ALIPAY = exports.URL_TYPE_PIS = exports.URL_TYPE_SOFORT = exports.URL_TYPE_SEPA_DIRECT_DEBIT_SECURED = exports.URL_TYPE_SEPA_DIRECT_DEBIT = exports.URL_TYPE_PRZELEWY24 = exports.URL_TYPE_PREPAYMENT = exports.URL_TYPE_PAYPAL = exports.URL_TYPE_INVOICE_SECURED = exports.URL_TYPE_INVOICE = exports.URL_TYPE_IDEAL = exports.URL_TYPE_GIROPAY = exports.URL_TYPE_EPS = exports.URL_TYPE_CARD = exports.URL_TYPE_BANCONTACT = exports.URL_RECURRING = exports.URL_BASKET = exports.URL_METADATA = exports.URL_CUSTOMER = exports.URL_PAYMENT_PAYOUT = exports.URL_PAYMENT_SHIPMENT = exports.URL_PAYMENT_CHARGE_CANCEL = exports.URL_PAYMENT_CHARGE_AUTHORIZE = exports.URL_PAYMENT_CHARGE = exports.URL_PAYMENT_AUTHORIZE_CANCEL = exports.URL_PAYMENT_AUTHORIZE = exports.URL_PAYMENT = void 0;
exports.URL_PAYMENT = '/payments';
exports.URL_PAYMENT_AUTHORIZE = exports.URL_PAYMENT + "/authorize";
exports.URL_PAYMENT_AUTHORIZE_CANCEL = exports.URL_PAYMENT + "/{paymentId}/authorize/{authorizationId}/cancels";
exports.URL_PAYMENT_CHARGE = exports.URL_PAYMENT + "/charges";
exports.URL_PAYMENT_CHARGE_AUTHORIZE = exports.URL_PAYMENT + "/{paymentId}/charges";
exports.URL_PAYMENT_CHARGE_CANCEL = exports.URL_PAYMENT + "/{paymentId}/charges/{chargeId}/cancels";
exports.URL_PAYMENT_SHIPMENT = exports.URL_PAYMENT + "/{paymentId}/shipments";
exports.URL_PAYMENT_PAYOUT = exports.URL_PAYMENT + "/payouts";
exports.URL_CUSTOMER = '/customers';
exports.URL_METADATA = '/metadata';
exports.URL_BASKET = '/baskets';
exports.URL_RECURRING = "/types/{typeId}/recurring";
exports.URL_TYPE_BANCONTACT = '/types/bancontact';
exports.URL_TYPE_CARD = '/types/card';
exports.URL_TYPE_EPS = '/types/eps';
exports.URL_TYPE_GIROPAY = '/types/giropay';
exports.URL_TYPE_IDEAL = '/types/ideal';
exports.URL_TYPE_INVOICE = '/types/invoice';
exports.URL_TYPE_INVOICE_SECURED = '/types/invoice-secured';
exports.URL_TYPE_PAYPAL = '/types/paypal';
exports.URL_TYPE_PREPAYMENT = '/types/prepayment';
exports.URL_TYPE_PRZELEWY24 = '/types/przelewy24';
exports.URL_TYPE_SEPA_DIRECT_DEBIT = '/types/sepa-direct-debit';
exports.URL_TYPE_SEPA_DIRECT_DEBIT_SECURED = '/types/sepa-direct-debit-secured';
exports.URL_TYPE_SOFORT = '/types/sofort';
exports.URL_TYPE_PIS = '/types/pis';
exports.URL_TYPE_ALIPAY = '/types/alipay';
exports.URL_TYPE_WECHATPAY = '/types/wechatpay';
exports.URL_TYPE_INSTALLMENT_SECURED = '/types/installment-secured';
exports.URL_TYPE_INSTALLMENT_SECURED_PLANS = '/types/installment-secured/plans';
exports.URL_PAYPAGE_CHARGE = '/paypage/charge';
exports.URL_PAYPAGE_AUTHORIZE = '/paypage/authorize';
exports.URL_WEBHOOK = '/webhooks';
exports.URL_LINKPAY = '/linkpay';
exports.URL_LINKPAY_CHARGE = '/linkpay/charge';
exports.URL_LINKPAY_AUTHORIZE = '/linkpay/authorize';
